simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -938,7 +938,14 @@ class simd<std::int32_t, simd_abi::avx2_fixed_size<4>> {
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_from(value_type const* ptr,
                                                        element_aligned_tag) {
+    // FIXME_HIP ROCm 5.6 can't compile with the intrinsic used here.
+#if defined(__HIPCC__) &&       \
+    ((HIP_VERSION_MAJOR > 5) || \
+     ((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR >= 6)))
```

Comment:
This is the kind of guards we don't want to read multiple times.  Consider doing this
```suggestion
#if defined(__HIPCC__) &&       \
    ((HIP_VERSION_MAJOR > 5) || \
     ((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR >= 6)))
#define KOKKOS_IMPL_WORKAROUND_ROCM_AVX2_ISSUE
#endif

#ifdef KOKKOS_IMPL_WORKAROUND_ROCM_AVX2_ISSUE
```
and `#undef` wherever appropriate.

Reply:
Sure. 65260e0ef37b496b7a95da49af62fd350e309127